src/shared/clients/s3Client.ts

```diff
@@ -264,6 +272,20 @@ export class DefaultS3Client {
         getLogger().debug('DownloadFile succeeded')
     }
 
+    public async getSignedUrl(request: SignedUrlRequest): Promise<string> {
+        const time = request.time ? request.time : 15
+        const operation = request.operation ? request.operation : 'getObject'
+        const s3 = await this.createS3()
+
+        const url = s3.getSignedUrl(operation, {
```

Comment:
This isn't synchronous, right? I think this would return a promise in that case. Try `getSignedUrlPromise` instead, and await the call.

Reply:
It is synchronous